flutter-studio/src/io/flutter/module/FlutterModuleModel.java

```diff
@@ -0,0 +1,150 @@
+    myModuleComponent = component;
+    myModuleName = new TextProperty(component.getModuleNameField());
+    myModuleContentRoot = new TextProperty(fetchPrivateField("myModuleContentRoot").getTextField());
+    myModuleFileLocation = new TextProperty(fetchPrivateField("myModuleFileLocation").getTextField());
+  }
+
+  @NotNull
+  public Project getProject() {
+    return myProject;
+  }
+
+  @NotNull
+  public TextProperty moduleName() {
+    return myModuleName;
+  }
+
+  @NotNull
+  public TextProperty moduleContentRoot() {
+    return myModuleContentRoot;
+  }
+
+  @NotNull
+  public TextProperty moduleFileLocation() {
+    return myModuleFileLocation;
+  }
+
+  @NotNull
+  public OptionalValueProperty<String> flutterSdk() {
+    return myFlutterSdk;
+  }
+
+  @Override
+  protected void handleFinished() {
+    // Set name, module file path, content entry path.
+    myModuleComponent.updateDataModel();
+    try {
+      // Create directories; this is not real validation.
+      if (!myModuleComponent.validate()) {
+        cleanupAfterValidationFailure();
+        Messages.showErrorDialog("Validation error", "Internal Error");
+        return;
+      }
+    }
+    catch (ConfigurationException e) {
+      // If there was a problem let the user see it to file a report.
+      // Re-throwing the exception brings up an error balloon with a 'Report to Google' button.
+      // Unfortunately, that button does not have any UI elements so we cannot customize the report.
+      cleanupAfterValidationFailure();
+      LOG.error(e);
+      Messages.showErrorDialog("Configuration error (stack trace logged):\n" + e.getMessage(), "Internal Error");
+      return;
+    }
+    if (myFlutterSdk.get().isPresent()) {
+      myBuilder.setFlutterSdkPath(myFlutterSdk.get().get());
+    }
+    else {
+      cleanupAfterValidationFailure(); // Not reached if validation is working.
+      return;
+    }
+    ModifiableModuleModel moduleModel = ModuleManager.getInstance(myProject).getModifiableModel();
+    final Exception[] error = new Exception[1];
+    ApplicationManager.getApplication().runWriteAction(() -> {
+      try {
+        myBuilder.createModule(moduleModel);
+      }
+      catch (ConfigurationException | IOException | ModuleWithNameAlreadyExists | JDOMException e) {
+        error[0] = e;
+      }
+    });
+    if (error[0] != null) {
+      cleanupAfterValidationFailure();
+      LOG.error(error[0]);
+      Messages.showErrorDialog("Module creation error (stack trace logged):\n" + error[0].getMessage(), "Internal Error");
+      return;
+    }
+    if (myBuilder.commitModule(getProject(), null) == null) {
+      // A message was displayed to the user, so just clean up.
+      cleanupAfterValidationFailure();
+    }
+  }
+
+  private void cleanupAfterValidationFailure() {
+    String filePath = myModuleContentRoot.get().trim();
+    //noinspection ResultOfMethodCallIgnored
+    new File(filePath).delete();
```

Comment:
I was going to say we should talk to the virtual filesystem to mediate this but on second thought that's probably not a good idea since there may be thread access issues.  In any event, the IDE has no business knowing about this file anyway. Which is just to say, feel free to ignore. 👍